app/controllers/users_controller.rb

```diff
@@ -51,12 +51,28 @@ def user_params
 
 class UsersController::InvitationsController < Devise::InvitationsController
   def get_user
-     User.find_by_email(resource_params[:email]) 
+     User.find_by_email(resource_params[:email])
   end
 
   include ElevatedPermissionsHelper
 
   def create
+    begin
+      params.require(:user).require(:email)
+    rescue ActionController::ParameterMissing
+      flash[:alert] = 'Email address is required'
+      redirect_to after_invite_path_for(current_inviter) and return
+    end
+    is_admin = current_inviter[:admin]
+    unless is_admin
+      begin
+        params.require(:user).require(:team_members_attributes).require('0').require('role')
+      rescue ActionController::ParameterMissing
+        flash[:alert] = 'Permission Title is required'
+        redirect_to after_invite_path_for(current_inviter) and return
+      end
+    end
+
```

Comment:
I agree with this part because it does the job but, it feels like it should be a candidate to be replaced with a 'validator' or a more ruby style validation. What do you think @arnau ?

Reply:
Yes, I agree in principal, I tried to use a Rails style validator i.e. as described in:  https://stackoverflow.com/a/32040081/238230 however, it appears that `role` is not part of the model and other things broke when I tried this approach.
I am happy to add it as a `TODO` as I think it would still be good to merge this as it is better than what we have currently.